### PR TITLE
bump testing for PT 2.1

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

https://dev-discuss.pytorch.org/t/pytorch-release-2-1-0/1271

<details>
  <summary>Before submitting</summary>

- [x] Was this **discussed/agreed** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/torchmetrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to **update the docs**?
- [ ] Did you write any new **necessary tests**?

</details>

<details>
  <summary>PR review</summary>

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

</details>

## Did you have fun?

Make sure you had fun coding 🙃

----
:books: Documentation preview :books:: https://torchmetrics--2142.org.readthedocs.build/en/2142/

### CODE REVIEW

**Comment by Borda** [comment]
let's wait for resolving the py3.11 in #2138

**Comment by Borda** [comment]
seems like some test just crash... :/

**Comment by stancld** [comment]
@Borda To me it looks like rather that PyTorch releases several new CPU + half implementations for functions like `sigmoid`, `sqrt`,.. So several calculations which are expected to fail now pass.

**Review by stancld — APPROVED** [review]

**Review by SkafteNicki — APPROVED** [review]

**Comment by Borda** [comment]
@SkafteNicki seems to be failing on Win + pt2.1
`unittests\image\test_perceptual_path_length.py::test_sim_net`

File: .github/workflows/ci-tests.yml [hunk 1]
```diff
@@ -32,20 +32,21 @@ jobs:
...
-          - { os: "macOS-12", python-version: "3.10", pytorch-version: "2.0.0" }
-          - { os: "macOS-12", python-version: "3.11", pytorch-version: "2.0.0" }
+          - { os: "macOS-12", python-version: "3.10", pytorch-version: "2.0.1" }
+          - { os: "macOS-12", python-version: "3.11", pytorch-version: "2.1.0" }
           - { os: "windows-2022", python-version: "3.8", pytorch-version: "1.13.1" }
           - { os: "windows-2022", python-version: "3.9", pytorch-version: "1.13.1" }
-          - { os: "windows-2022", python-version: "3.10", pytorch-version: "2.0.0" }
-          - { os: "windows-2022", python-version: "3.11", pytorch-version: "2.0.0" }
+          - { os: "windows-2022", python-version: "3.10", pytorch-version: "2.0.1" }
+          - { os: "windows-2022", python-version: "3.11", pytorch-version: "2.1.0" }
```
Comment (Borda) [inline .github/workflows/ci-tests.yml:49]:
```suggestion
          - { os: "windows-2022", python-version: "3.10", pytorch-version: "2.1.0" }
```

**Pull request merged**